src/mlpack/methods/ann/loss_functions/bce_with_logits_loss.hpp

```diff
@@ -0,0 +1,130 @@
+/**
+ * @file bce_with_logits_loss.hpp
+ * @author Mrityunjay Tripathi
+ *
+ * Definition of the Binary Cross Entropy with Logits Loss function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LOSS_FUNCTIONS_BCE_WITH_LOGITS_LOSS_HPP
+#define MLPACK_METHODS_ANN_LOSS_FUNCTIONS_BCE_WITH_LOGITS_LOSS_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The binary cross-entropy with logits performance function measures the loss
+ * between the input and target distributions, where the labels in target can
+ * be 0 or 1. This loss combines a Sigmoid layer and the BCELoss in one single
+ * class. This version is more numerically stable than using a plain Sigmoid
+ * followed by a BCELoss as, by combining the operations into one layer,
+ * we take advantage of the log-sum-exp trick for numerical stability.
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class BCEWithLogitsLoss
+{
+ public:
+  /**
+   * Create the BCEWithLogitsLoss object.
+   *
+   * @param weight The manual rescaling factor given to loss.
+   * @param posWeight The weight of positive of examples.
+   *    posWeight = total negative examples / total positive examples.
```

Comment:
This almost reads like it's required that posWeight be set as such. I'm not sure how to convey this succinctly, but we could describe this as describing a good initial value for posWeight. Though, maybe only I'm reading it like this.